File: V семестр/Паралельне програмування/Бровченко/lab3_OPP/lab3_OPP/lab3_OPP/F1.cs
using System;

namespace lab3
{
    public class F1
    {
    private Vector result;
    public Vector getResult()
    {
        return result;
    }

    private int N;

    public F1(int N)
    {
        this.N = N;
    }

    public void run()
    {
            Console.WriteLine("Task 1 start");
            Vector A = new Vector(N), B = new Vector(N);
            Matrix MA = new Matrix(N), MC = new Matrix(N);
            long e = 3;
            result = A.sub(MA.multiply(MC).multiply(B.multiply(e)));
            Console.WriteLine("Task 1 end");
    }
}

}

File: V семестр/Паралельне програмування/Бровченко/lab2_ОРP/src/F1.java
/**
 * F1: C = A-B*(MA*MC)*e
 */


public class F1 extends Thread {

    int[] A, B;
    int e;
    int[][] MA, MC;
    private int[] result;
    private Data d;

    F1(String name, int priority, Data d) {
        setName(name);
        setPriority(priority);
        this.d = d;
        }

    @Override
    public void run() {
        try {
            System.out.println("Task 1 start");
            int n = d.getN();
            int[] A = new int[n], B = new int[n];
            int e = 3;
            int[][] MA = new int [n][n], MC = new int[n][n];
            result = d.func1(A, B, MA, MC, e);
            sleep(500);
            System.out.println("Task 1 end");
        } catch (InterruptedException e) {
            e.printStackTrace();
        }
    }
}
